### DISCUSSION TITLE
How do I get ng command to work in Codespaces?

### DISCUSSION THREAD

**Discussion by abhidotnet** [community/community#27184] [Codespaces]
<p>Does the ng command work in Codespaces? I have just started playing with the beta and am wondering how to get my code to actually compile as it does in my VS Code environment.</p>

**Comment by edwinhollen** [comment, marked as answer]
<p>Hi Abhijit, the Angular CLI isn’t installed by default; you need to install the Angular CLI using the terminal feature in the Codespace:</p>
<ol>
<li>Open the Menu &gt; Terminal &gt; New Terminal</li>
<li>Run <code>npm install -g @angular/cli</code>
</li>
</ol>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      
      <a href="https://angular.io/cli" target="_blank" rel="noopener nofollow ugc">angular.io</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://angular.io/cli" target="_blank" rel="noopener nofollow ugc">Angular</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Hope that helps</p>

**Comment by anirugu** [comment]
<p>I tried to install the angular cli but it doesn’t look like working. Please check </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/d/bd6f5964403a98c17468a3688d7053103780beb9.png" data-download-href="/uploads/short-url/r1ONZYydhC92BMC7d5f0U1SmNux.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/35772421/181102178-65fdcd5a-4e33-4b78-aeb7-749353881d1a.png" alt="image" data-base62-sha1="r1ONZYydhC92BMC7d5f0U1SmNux" width="690" height="262" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd6f5964403a98c17468a3688d7053103780beb9_2_690x262.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd6f5964403a98c17468a3688d7053103780beb9_2_1035x393.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd6f5964403a98c17468a3688d7053103780beb9_2_1380x524.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd6f5964403a98c17468a3688d7053103780beb9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1527×581 48.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Chuxel** [comment]
<p>Thanks for reporting this.  It looks like there’s a bug in the PATH.  Run <code>export PATH=${PATH}:/home/codespace/.npm-global/bin</code> and it will work.</p>
<p>You can also cause this to always execute by running the following:</p>
<p><code>echo 'export PATH=${PATH}:/home/codespace/.npm-global/bin' | tee -a ~/.bashrc &gt;&gt; ~/.zshrc</code></p>

**Comment by jkeech** [comment]
<p>This is fixed in this PR – it should be deployed to Codespaces over the next couple days: <a href="https://github.com/microsoft/vscode-dev-containers/pull/686">https://github.com/microsoft/vscode-dev-containers/pull/686</a></p>
